x-pack/plugins/apm/public/components/app/infra_overview/infra_tabs/index.tsx

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  EuiTabbedContent,
+  EuiTabbedContentProps,
+  EuiLoadingSpinner,
+} from '@elastic/eui';
+import React, { useMemo } from 'react';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import type { QueryDslQueryContainer } from '@elastic/elasticsearch/lib/api/types';
+import { useApmServiceContext } from '../../../../context/apm_service/use_apm_service_context';
+import { useApmParams } from '../../../../hooks/use_apm_params';
+import { useTimeRange } from '../../../../hooks/use_time_range';
+import { FETCH_STATUS, useFetcher } from '../../../../hooks/use_fetcher';
+import { ApmPluginStartDeps } from '../../../../plugin';
+import { EmptyPrompt } from './empty_prompt';
+import { FailurePrompt } from './failure_prompt';
+
+type Tab = NonNullable<EuiTabbedContentProps['tabs']>[0] & {
+  id: 'containers' | 'pods' | 'hosts';
+  hidden?: boolean;
+};
+
+export function InfraTabs() {
+  const { serviceName } = useApmServiceContext();
+  const {
+    query: { environment, kuery, rangeFrom, rangeTo },
+  } = useApmParams('/services/{serviceName}/infra');
+  const { start, end } = useTimeRange({ rangeFrom, rangeTo });
+
+  const { data, status } = useFetcher(
```

Comment:
You can add an `INITIAL_STATE` here to be used when data is undefined, like https://github.com/elastic/kibana/blob/main/x-pack/plugins/apm/public/components/app/service_map/popover/service_contents.tsx#L60